src/core/library.c

```diff
@@ -663,6 +663,54 @@ QuicLibraryGetGlobalParam(
         Status = QUIC_STATUS_SUCCESS;
         break;
 
+    case QUIC_PARAM_GLOBAL_PERF_COUNTERS: {
+
+        if (*BufferLength < sizeof(uint64_t)) {
+            *BufferLength = sizeof(uint64_t) * QUIC_PERF_COUNTER_MAX;
+            Status = QUIC_STATUS_BUFFER_TOO_SMALL;
+            break;
+        }
+
+        if (Buffer == NULL) {
+            Status = QUIC_STATUS_INVALID_PARAMETER;
+            break;
+        }
+
+        uint64_t MergedPerfCounters[QUIC_PERF_COUNTER_MAX];
```

Comment:
Don't use the temporary variable. Use the output buffer directly. Then, instead of looping over `QUIC_PERF_COUNTER_MAX` counters when summing, loop over `*BufferLength / sizeof(uint64_t)`'s worth. Then you don't need to copy over at the end.